src/mac.cpp

```diff
@@ -154,6 +154,57 @@ CFArrayRef CreateChildrenArray(AXUIElementRef element) {
   return children;
 }
 
+CFStringRef GetNestedText(AXUIElementRef element, CFMutableArrayRef textChildren) {
+  if (element == NULL) {
+    return NULL;
+  }
+
+  CFIndex count;
+  CFArrayRef children;
+  AXUIElementGetAttributeValueCount(element, kAXChildrenAttribute, &count);
+
+  if (count == 0) {
+    CFStringRef value;
+    AXUIElementCopyAttributeValue(element, kAXValueAttribute, reinterpret_cast<CFTypeRef*>(&value));
+    CFArrayAppendValue(textChildren, value);
+  } else {
+    AXUIElementCopyAttributeValues(element, kAXChildrenAttribute, 0, count, &children);
+    for (CFIndex i = 0; i < count; i++) {
+      AXUIElementRef child = static_cast<AXUIElementRef>(CFArrayGetValueAtIndex(children, i));
+      GetNestedText(child, textChildren);
+    }
+  }
+  CFStringRef combined = CFStringCreateByCombiningStrings(kCFAllocatorDefault, textChildren, CFSTR(""));
+  return combined;
```

Comment:
I think I've released every object that needs it now. I handle the string casting and memory management in `GetEditorState` so that I can call this function recursively and also separate out the weird encoding things that need to be done later.